x-pack/plugins/canvas/types/arguments.ts

```diff
@@ -54,3 +56,57 @@ export interface ArgumentConfig<Arguments = {}> {
   /** A description of the argument */
   help?: string;
 }
+
+// This will be the props for FunctionForm component and ends up being passed into
+// FunctionForm model's render function
+// TODO: Continue to build this out as needed
+export interface FunctionFormRenderArgs<ContextValueType> {
+  context: ExpressionContext<ContextValueType>;
+}
+
+type AnyExpressionFunctionDefinition = ExpressionFunctionDefinition<any, any, any, any>;
+
+export type ArgumentTypeDefinitionArgument<Arguments> = ArgumentConfig<Arguments> & {
+  options?: Record<string, any>;
+};
+
+type ExtractExpressionFunctionArguments<E> = E extends ExpressionFunctionDefinition<
```

Comment:
This file definitely needs some comments... it will help folks like me follow what these types are doing.